composeApp/src/commonMain/kotlin/sp/bvantur/tasky/core/data/NetworkUtils.kt

```diff
@@ -2,36 +2,50 @@ package sp.bvantur.tasky.core.data
 
 import io.ktor.client.call.body
 import io.ktor.client.plugins.ClientRequestException
+import io.ktor.client.plugins.RedirectResponseException
 import io.ktor.client.plugins.ServerResponseException
 import io.ktor.client.statement.HttpResponse
+import io.ktor.http.HttpStatusCode
 import io.ktor.http.isSuccess
+import io.ktor.util.network.UnresolvedAddressException
 import kotlinx.coroutines.TimeoutCancellationException
-import kotlinx.io.IOException
-import sp.bvantur.tasky.core.domain.CommunicationError
+import sp.bvantur.tasky.core.domain.TaskyError
 import sp.bvantur.tasky.core.domain.TaskyResult
 import kotlin.coroutines.cancellation.CancellationException
 
-suspend inline fun <reified T> safeApiCall(execute: () -> HttpResponse): TaskyResult<T, CommunicationError> {
-    // TODO revisit and handle more cases here
-    return try {
-        val response = execute()
-        if (response.status.isSuccess()) {
-            val responseBody = response.body<T>()
-            TaskyResult.Success(responseBody)
-        } else {
-            TaskyResult.Error(CommunicationError.UnknownError())
-        }
-    } catch (e: ClientRequestException) {
-        return TaskyResult.Error(CommunicationError.UnknownError(e.message))
-    } catch (e: ServerResponseException) {
-        return TaskyResult.Error(CommunicationError.UnknownError(e.message))
-    } catch (e: IOException) {
-        return TaskyResult.Error(CommunicationError.UnknownError(e.message))
-    } catch (e: TimeoutCancellationException) {
-        return TaskyResult.Error(CommunicationError.UnknownError(e.message))
-    } catch (ignore: Exception) {
-        if (ignore is CancellationException) throw ignore
+suspend inline fun <reified T> safeApiCall(execute: () -> HttpResponse): TaskyResult<T, TaskyError> = try {
+    val response = execute()
 
-        return TaskyResult.Error(CommunicationError.UnknownError(ignore.message))
+    if (response.status.isSuccess()) {
+        val responseBody = response.body<T>()
+        TaskyResult.Success(responseBody)
+    } else {
+        response.toResult()
     }
+} catch (ignore: RedirectResponseException) {
+    TaskyResult.Error(TaskyError.RedirectError)
+} catch (ignore: ClientRequestException) {
+    TaskyResult.Error(TaskyError.BadRequest)
+} catch (ignore: ServerResponseException) {
+    TaskyResult.Error(TaskyError.ServerError)
+} catch (ignore: UnresolvedAddressException) {
+    TaskyResult.Error(TaskyError.NoInternet)
+} catch (ignore: TimeoutCancellationException) {
+    TaskyResult.Error(TaskyError.RequestTimeout)
+} catch (ignore: kotlinx.io.IOException) {
+    TaskyResult.Error(TaskyError.NoInternet)
+} catch (ignore: Exception) {
```

Comment:
minor - is there any specific reason why you named this `ignore`? It does not seem like it would represent an exception object